test/node/NodeMetricsProvider.spec.ts

```diff
@@ -81,43 +84,52 @@ describe('NodeMetricsProvider', function () {
       ensureCall({ spy: storageProviderStub.setStore, callCount: 0 })
     })
 
-    it('User updates consent', function () {
+    it('User updates consent', async function () {
       const storedConsent: consentTypes[] = ['minimal']
       storageProviderStub.getStore.returns(storedConsent)
       const telemetry = new NodeMetricsProvider({ appKey: 'foo', url: 'bar', storageProvider: storageProviderStub })
+      while (!telemetry.initDone) {
+        await new Promise(resolve => setTimeout(resolve, 10))
+      }
       expect(telemetry).to.have.property('storageProvider').that.is.not.null
 
       ensureCall({ spy: storageProviderStub.getStore, callCount: 1 })
       ensureCall({ spy: countlyStub.add_consent, callCount: 1, callIndex: 0, expectedArgs: storedConsent })
       ensureCall({ spy: storageProviderStub.setStore, callCount: 0 })
-      telemetry.addConsent('performance')
+      await telemetry.addConsent('performance')
       ensureCall({ spy: storageProviderStub.getStore, callCount: 1 })
       ensureCall({ spy: countlyStub.add_consent, callCount: 2, callIndex: 1, expectedArgs: ['performance'] })
       ensureCall({ spy: storageProviderStub.setStore, callCount: 1, callIndex: 0, expectedArgs: ['minimal', 'performance'] })
     })
 
-    it('User has multiple consents stored', function () {
+    it('User has multiple consents stored', async function () {
       const storedConsent: consentTypes[] = ['minimal', 'performance']
       storageProviderStub.getStore.returns(storedConsent)
       const telemetry = new NodeMetricsProvider({ appKey: 'foo', url: 'bar', storageProvider: storageProviderStub })
+      while (!telemetry.initDone) {
+        await new Promise(resolve => setTimeout(resolve, 10))
+      }
       expect(telemetry).to.have.property('storageProvider').that.is.not.null
 
       ensureCall({ spy: storageProviderStub.getStore, callCount: 1 })
       ensureCall({ spy: countlyStub.add_consent, callCount: 1, callIndex: 0, expectedArgs: storedConsent })
       ensureCall({ spy: storageProviderStub.setStore, callCount: 0 })
     })
 
-    it('User removes consent', function () {
+    it('User removes consent', async function () {
       const storedConsent: consentTypes[] = ['minimal']
       storageProviderStub.getStore.returns(storedConsent)
       const telemetry = new NodeMetricsProvider({ appKey: 'foo', url: 'bar', storageProvider: storageProviderStub })
+      while (!telemetry.initDone) {
+        await new Promise(resolve => setTimeout(resolve, 10))
+      }
```

Comment:
not a big fan of needing a while loop, but im okay with it for now.

Can we add a followup action item for exposing a promise that folks can await to determine when init is done?